eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventData.java

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.eventhubs.implementation.AmqpConstants;
+import com.azure.eventhubs.implementation.EventDataUtil;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.Binary;
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.amqp.messaging.Data;
+import org.apache.qpid.proton.amqp.messaging.MessageAnnotations;
+import org.apache.qpid.proton.message.Message;
+
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.azure.eventhubs.implementation.AmqpConstants.ENQUEUED_TIME_UTC_ANNOTATION_NAME;
+import static com.azure.eventhubs.implementation.AmqpConstants.OFFSET_ANNOTATION_NAME;
+import static com.azure.eventhubs.implementation.AmqpConstants.PARTITION_KEY_ANNOTATION_NAME;
+import static com.azure.eventhubs.implementation.AmqpConstants.PUBLISHER_ANNOTATION_NAME;
+import static com.azure.eventhubs.implementation.AmqpConstants.SEQUENCE_NUMBER_ANNOTATION_NAME;
+
+/**
+ * The data structure encapsulating the Event being sent-to and received-from EventHubs.
```

Comment:
That's a typo. I'll fix it.

The product is Event Hubs, but when you create an EventHubClient, it can only be connected to a single Event Hub. 🤣 So there's a bunch of places I dropped the plural because it refers to a single hub.